auth/auth.go

```diff
@@ -116,10 +127,10 @@ func (p Auth) getRoleARN() (arn *string, e error) {
 	}
 
 	roleArn := rsp.Annotations[arnAnno]
-	if len(roleArn) <= 0 {
-		klog.Errorf("Need IAM role for service account %s (namespace: %s) - %s", p.svcAcc, p.nameSpace, docURL)
-		return nil, fmt.Errorf("An IAM role must be associated with service account %s (namespace: %s)", p.svcAcc, p.nameSpace)
-	}
+	// if len(roleArn) <= 0 {
```

Comment:
Did you mean to delete this?

Reply:
yes ..otherwise we need to have some other condition to know this pod use IRSA or PIA 
Current logic is if role-arn annotations in ServiceAcccount is empty then assume its PIA else IRSA